optimum/intel/openvino/modeling_open_clip.py

```diff
@@ -0,0 +1,733 @@
+            trust_remote_code=trust_remote_code,
+            task=task,
+            **kwargs,
+        )
+
+
+@add_start_docstrings(
+    """
+    OpenVINO Model for OpenCLIP Text model for tasks like zero-shot-image-classification.
+    """,
+    MODEL_START_DOCSTRING,
+)
+class OVModelOpenCLIPText(OVModelOpenCLIPBase):
+    OV_XML_FILE_NAME = "openvino_model_text.xml"
+
+    def __init__(self, model=None, config=None, tokenize_cfg=None, **kwargs):
+        super().__init__(model, config, **kwargs)
+        self.tokenize_cfg = tokenize_cfg
+
+    @classmethod
+    def _from_transformers(
+        cls,
+        model_id: str,
+        config: PretrainedConfig,
+        use_auth_token: Optional[Union[bool, str]] = None,
+        token: Optional[Union[bool, str]] = None,
+        revision: Optional[str] = None,
+        force_download: bool = False,
+        cache_dir: str = HUGGINGFACE_HUB_CACHE,
+        subfolder: str = "",
+        local_files_only: bool = False,
+        task: Optional[str] = None,
+        trust_remote_code: bool = False,
+        load_in_8bit: Optional[bool] = None,
+        quantization_config: Union[OVWeightQuantizationConfig, Dict] = None,
+        **kwargs,
+    ):
+        if use_auth_token is not None:
+            warnings.warn(
+                "The `use_auth_token` argument is deprecated and will be removed soon. Please use the `token` argument instead.",
+                FutureWarning,
+            )
+            if token is not None:
+                raise ValueError("You cannot use both `use_auth_token` and `token` arguments at the same time.")
+            token = use_auth_token
+
+        save_dir = TemporaryDirectory()
+        save_dir_path = Path(save_dir.name)
+        # This attribute is needed to keep one reference on the temporary directory, since garbage collecting
+        # would end-up removing the directory containing the underlying OpenVINO model
+        cls._model_save_dir_tempdirectory_instance = save_dir
+
+        # If load_in_8bit and quantization_config not specified then ov_config is set to None and will be set by default in convert depending on the model size
+        if load_in_8bit is None and not quantization_config:
+            ov_config = None
+        else:
+            ov_config = OVConfig(dtype="fp32")
+
+        def fn_get_submodels(model):
+            return {"model_text": model.text}
+
+        custom_export_configs = {
+            "model_text": {},
+        }
+
+        main_export(
+            model_name_or_path=model_id,
+            output=save_dir_path,
+            task=task or "zero-shot-image-classification",
+            subfolder=subfolder,
+            revision=revision,
+            cache_dir=cache_dir,
+            token=token,
+            local_files_only=local_files_only,
+            force_download=force_download,
+            trust_remote_code=trust_remote_code,
+            ov_config=ov_config,
+            library_name="open_clip",
```

Comment:
can be set as `_library_name` https://github.com/huggingface/optimum-intel/blob/ba74307dfc7f69c6281e811e02ee0203a6fb8355/optimum/intel/openvino/modeling_base.py#L63